src/org/omegat/core/search/Searcher.java

```diff
@@ -174,9 +174,15 @@ public void search() throws TranslationException, PatternSyntaxException, IOExce
         switch (expression.searchExpressionType) {
         case EXACT:
```

Comment:
It is too hard for me to implement it. Because method should recognize regex expression and produce valid expression. it is not valid only with skipping chars that used for regexp.
```
Ex.
 "^\t[abc]+$"  become like "^t[abc\uff41\uff42\uff43]+$"
"^\t\$[abc]+$" become like "^\t\$\uff04[abc\uff41\uff42\uff43]+$"
"\p{javaWhitespace}"  keep it
```

It may become like an implementation of regex parser to support REGEXP mode and full-width and half-width insensitive search.